metapool/tests/test_plate.py

```diff
@@ -563,6 +564,140 @@ def test_sample_plate_has_no_spaces(self):
         self.assertEqual(context, expected)
 
 
+class PlateReplicationTests(TestCase):
+    def test_overwrite_source_quad(self):
+        # replicate a valid source to empty sources 2 and 4 plus overwriting
+        # source 3.
+
+        input_df = pd.read_csv('metapool/tests/data/input_plate.tsv',
+                               sep='\t', dtype=str)
+
+        pr = PlateReplication('Library Well')
+
+        obs = pr.make_replicates(input_df, {1: [2, 3, 4]}, overwrite=True)
+
+        exp = pd.read_csv('metapool/tests/data/file1.tsv',
+                          sep='\t', dtype=str)
+
+        pd.set_option('display.max_columns', None)
+
+        obs = obs.set_index('Sample').to_dict(orient='index')
+        exp = exp.set_index('Sample').to_dict(orient='index')
+
+        sample_names = set(obs.keys()).union(set(exp.keys()))
+
+        for sample_name in sample_names:
+            self.assertDictEqual(obs[sample_name], exp[sample_name])
+
+        # replicate a valid source to empty sources 2 and 4 plus overwriting
+        # source 3 with overwrites not allowed. Should return an Error.
+
+        with self.assertRaisesRegex(ValueError, 'Quadrant 3 is a source '
+                                                'quadrant'):
+            pr.make_replicates(input_df, {1: [2, 3, 4]}, overwrite=False)
```

Comment:
Is there a reason to not use here pandas.testing.assert_series_equal ?

Reply:
This also applies to all the tests in this file.

Reply:
Ty for the tip! I ended up using assert_frame_equal().